bemserver_core/model/campaigns.py

```diff
@@ -152,3 +134,33 @@ def register_class(cls):
                 ),
             },
         )
+
+
+def init_db_campaigns_triggers():
+    """Create triggers to protect some columns from update.
+
+    This function is meant to be used for tests or dev setups after create_all.
+    Production setups should rely on migration scripts.
+    """
+
+    # Set "update read-only trigger" on campaign_id column for CampaignScope table.
+    db.session.execute(
+        sqla.DDL(
+            f"""CREATE TRIGGER
+    {CampaignScope.__table__}_trigger_bu_campaign_not_allowed
+BEFORE UPDATE
+    OF {CampaignScope.campaign_id.key} ON {CampaignScope.__table__}
+FOR EACH ROW
+    WHEN (
+        NEW.{CampaignScope.campaign_id.key}
+        IS DISTINCT FROM
+        OLD.{CampaignScope.campaign_id.key}
+    )
+    EXECUTE FUNCTION column_update_not_allowed(
+        {CampaignScope.campaign_id.key},
+        {CampaignScope.name.key}
+    );"""
+        )
+    )
```

Comment:
There must be room for factorization. Couldn't we write a helper function to do this and call it with the model class and the read-only attribute name(s)?